Assets/Scripts/Models/World.cs

```diff
@@ -342,6 +342,44 @@ public Utility PlaceUtility(string objectType, Tile tile, bool doRoomFloodFill =
         return PlaceUtility(util, tile, doRoomFloodFill);
     }
 
+    /// <summary>
+    /// Reserves the furniture's work spot, preventing it from being built on.
+    /// </summary>
+    /// <param name="furn">The furniture whose workspot will be reserved.</param>
+    /// <param name="tile">The tile on which the furniture is located, for furnitures which don't have a tile, such as prototypes.</param>
+    public void ReserveTileAsWorkSpot(Furniture furn, Tile tile = null)
+    {
+        if (tile == null)
+        {
+            tile = furn.Tile;
+        }
+
+        World.Current.GetTileAt(
```

Comment:
We should create a function for this in World. A similar code is used in furniture.Jobs, and a few times here.